src/server/conn_context.cc

```diff
@@ -6,12 +6,39 @@
 
 #include "base/logging.h"
 #include "server/engine_shard_set.h"
+#include "server/server_family.h"
+#include "server/server_state.h"
+#include "src/facade/dragonfly_connection.h"
 #include "util/proactor_base.h"
 
 namespace dfly {
 
 using namespace std;
 
+void ConnectionContext::SendMonitorMsg(std::string msg) {
+  CHECK(owner());
+
+  owner()->SendMonitorMsg(std::move(msg));
+}
+
+void ConnectionContext::ChangeMonitor(bool start) {
+  // This will either remove or register a new connection
+  // at the "top level" thread --> ServerState context
+  if (start) {
+    conn_state.monitor.reset(new ConnectionState::Monitor);
+    MonitorsRepo::MonitorInfo new_entry(this);
+    shard_set->pool()->Await(
+        [&new_entry](auto*) { ServerState::tlocal()->Monitors().Add(new_entry); });
```

Comment:
Ok, I think it can be simplified. You actually never access this entry from another thread, so what you really want to register is a thread id of the connection with all the threads, IMHO.
and when you send monitor msg, you want to go over all the monitor connections in their own thread.

Reply:
in fact, all we actually need to store in ServerState is `uint32_t num_monitor_connections`, IMHO.
Because you broadcast the monitor message anyway to all the threads.

Reply:
This is what it dose, The registration is for the thread id of the connection that you send monitor messages on. Then when you have a new command, you need to make sure that you're sending the message in the right thread context, since you have in this list connections from multiple threads. I didn't tried to optimised this since it is assumed to have very few if any connections here.

Reply:
this is going to change. I will remove this, and change the logic for this.